indexer/sql/src/active_model.rs

```diff
@@ -0,0 +1,95 @@
+use {
+    crate::entity,
+    grug_math::Inner,
+    grug_types::{BlockInfo, Tx, TxOutcome},
+    sea_orm::{prelude::*, sqlx::types::chrono::TimeZone, Set},
+};
+
+#[derive(Debug, Default)]
+pub struct Models {
+    pub block: entity::blocks::ActiveModel,
+    pub transactions: Vec<entity::transactions::ActiveModel>,
+    pub messages: Vec<entity::messages::ActiveModel>,
+    pub events: Vec<entity::events::ActiveModel>,
+}
+
+impl Models {
+    pub fn push(&mut self, tx: &Tx, tx_outcome: &TxOutcome) {
+        let transaction_id = Uuid::new_v4();
+        let sender = tx.sender.to_string();
+        let new_transaction = entity::transactions::ActiveModel {
+            id: Set(transaction_id),
+            has_succeeded: Set(tx_outcome.result.is_ok()),
+            error_message: Set(tx_outcome.clone().result.err()),
+            gas_wanted: Set(tx.gas_limit.try_into().unwrap()),
+            gas_used: Set(tx_outcome.gas_used.try_into().unwrap()),
+            created_at: self.block.created_at.clone(),
+            block_height: self.block.block_height.clone(),
+            hash: Set("".to_string()),
+            data: Set(tx.data.clone().into_inner()),
+            sender: Set(sender),
+            credential: Set(tx.credential.clone().into_inner()),
+        };
+        self.transactions.push(new_transaction);
+
+        for message in tx.msgs.iter() {
+            let serialized_message = serde_json::to_value(message).unwrap();
```

Comment:
you can use trait `JsonSerExt` from `grug_types` instead tt maintain consistency with others crates
```suggestion
            let serialized_message = message.to_json_value()?;
```